plan/logical_plan_builder.go

```diff
@@ -1342,6 +1350,52 @@ func (b *planBuilder) checkOnlyFullGroupBy(p LogicalPlan, fields []*ast.SelectFi
 	}
 }
 
+func (b *planBuilder) checkOnlyFullGroupByWithOutGroupClause(p LogicalPlan, fields []*ast.SelectField) {
+	resolver := colResolverForOnlyFullGroupBy{}
+	for idx, field := range fields {
+		resolver.exprIdx = idx
+		field.Accept(&resolver)
+		if err := resolver.Check(); err != nil {
+			b.err = err
+			return
+		}
+	}
+}
+
+// colResolverForOnlyFullGroupBy visits Expr tree to find out if an Expr tree is an aggregation function.
+// If so, find out the first column name that not in an aggregation function.
+type colResolverForOnlyFullGroupBy struct {
+	firstNonAggCol    *ast.ColumnName
+	exprIdx           int
+	firstNonAggColIdx int
+	hasAggFunc        bool
+}
+
+func (c *colResolverForOnlyFullGroupBy) Enter(node ast.Node) (ast.Node, bool) {
+	switch node.(type) {
+	case *ast.AggregateFuncExpr:
+		c.hasAggFunc = true
+		return node, true
+	case *ast.ColumnNameExpr:
+		if c.firstNonAggCol == nil {
+			c.firstNonAggCol, c.firstNonAggColIdx = node.(*ast.ColumnNameExpr).Name, c.exprIdx
+		}
+		return node, true
+	}
+	return node, false
+}
+
+func (c *colResolverForOnlyFullGroupBy) Leave(node ast.Node) (ast.Node, bool) {
+	return node, true
+}
+
+func (c *colResolverForOnlyFullGroupBy) Check() error {
+	if c.hasAggFunc && c.firstNonAggCol != nil {
```

Comment:
How about `select min(c1) from t where c2 in (select c2 from t where c3 > 10);` Could this checker handle the situation of subquery?

Reply:
No, this situation is not handled by this checker. ColumnRefs in predicates is not affected under this circumstance because predicates are processed prior to aggregations.

Reply:
At least, this PR is an improvement. @shenli